Mathlib/Analysis/NormedSpace/MultipliableUniformlyOn.lean

```diff
@@ -0,0 +1,125 @@
+    fun y hy a b hab => (hab.symm ▸ hy :)⟩
+  filter_upwards [eventually_prod_principal_iff.mp hf2] with i hi x hx
+  simpa using add_lt_add_of_le_of_lt (hg x hx) (hi x hx)
+
+lemma TendstoUniformlyOn.eventually_forall_le {f : ι → α → β} {p : Filter ι} {g : α → β}
+    {K : Set α} {u v : β} (huv : u < v) (hf : TendstoUniformlyOn f g p K) (hg : ∀ x ∈ K, g x ≤ u) :
+    ∀ᶠ i in p, ∀ x ∈ K, f i x ≤ v := by
+  filter_upwards [hf.eventually_forall_lt huv hg] with i hi x hx using (hi x hx).le
+
+lemma TendstoUniformlyOn.comp_cexp {p : Filter ι} {f : ι → α → ℂ} {g : α → ℂ}
+    {K : Set α} (hf : TendstoUniformlyOn f g p K) (hg : BddAbove <| (fun x ↦ (g x).re) '' K) :
+    TendstoUniformlyOn (cexp ∘ f ·) (cexp ∘ g) p K := by
+  obtain ⟨v, hv⟩ : ∃ v, ∀ x ∈ K, (g x).re ≤ v := hg.imp fun _ h ↦ by simpa [mem_upperBounds] using h
+  have : ∀ᶠ i in p, ∀ x ∈ K, (f i x).re ≤ v + 1 := hf.re.eventually_forall_le (lt_add_one v) hv
+  refine (UniformContinuousOn.cexp _).comp_tendstoUniformlyOn_eventually (by simpa) ?_ hf
+  simpa using fun a ha ↦ (hv a ha).trans (lt_add_one v).le
+
+lemma Complex.HasSumUniformlyOn_log_one_add {α : Type*} {f : ι → α → ℂ} (K : Set α)
+    {u : ι → ℝ} (hu : Summable u) (h : ∀ᶠ n in cofinite, ∀ x ∈ K, ‖f n x‖ ≤ u n) :
+    HasSumUniformlyOn (fun i a ↦ log (1 + f i a)) (fun a ↦ ∑' i, log (1 + f i a)) {K} := by
+  simp only [hasSumUniformlyOn_iff_tendstoUniformlyOn, Set.mem_singleton_iff, forall_eq]
+  apply tendstoUniformlyOn_tsum_of_cofinite_eventually <| hu.mul_left (3 / 2)
+  filter_upwards [h, hu.tendsto_cofinite_zero.eventually_le_const one_half_pos] with i hi hn' x hx
+    using (norm_log_one_add_half_le_self <| (hi x hx).trans hn').trans (by simpa using hi x hx)
+
+lemma Complex.tendstoUniformlyOn_tsum_nat_log_one_add {α : Type*} {f : ℕ → α → ℂ} (K : Set α)
+    {u : ℕ → ℝ} (hu : Summable u) (h : ∀ᶠ n in atTop, ∀ x ∈ K, ‖f n x‖ ≤ u n) :
+    TendstoUniformlyOn (fun n a => ∑ i ∈ Finset.range n,
+    (Complex.log (1 + f i a))) (fun a => ∑' i : ℕ, Complex.log (1 + f i a)) atTop K := by
+  rw [← Nat.cofinite_eq_atTop] at h
+  simpa only [Set.mem_singleton_iff, forall_eq] using
+    hasSumUniformlyOn_tendstoUniformlyOn_nat (Complex.HasSumUniformlyOn_log_one_add K hu h)
+
+end cexp_clog
+
+section UniformlyOn
+
+lemma HasProdUniformlyOn_of_clog {f : ι → α → ℂ} {𝔖 : Set (Set α)}
+    (hf : SummableUniformlyOn (fun i a => log (f i a)) 𝔖) (hfn : ∀ K ∈ 𝔖, ∀ x ∈ K, ∀ i, f i x ≠ 0)
+    (hg : ∀ K ∈ 𝔖, BddAbove <| (fun x => (∑' n : ι, log (f n x)).re) '' K) :
+    HasProdUniformlyOn f (fun a => ∏' i, f i a) 𝔖 := by
+  simp only [hasProdUniformlyOn_iff_tendstoUniformlyOn, Set.mem_singleton_iff, forall_eq]
+  obtain ⟨r, hr⟩ := hf.exists
+  intro K hK
+  suffices H : TendstoUniformlyOn (fun s a => ∏ i ∈ s, f i a) (cexp ∘ r) atTop K by
+        apply TendstoUniformlyOn.congr_right H
+        apply Set.EqOn.trans (Set.EqOn.comp_left (hr.tsum_eqOn hK)).symm
+        exact fun x hx ↦ (cexp_tsum_eq_tprod (hfn K hK x hx) (hf.summable hK hx))
```

Comment:
```suggestion
  apply TendstoUniformlyOn.congr_right H
  apply Set.EqOn.trans (Set.EqOn.comp_left (hr.tsum_eqOn hK)).symm
  exact fun x hx ↦ (cexp_tsum_eq_tprod (hfn K hK x hx) (hf.summable hK hx))
```

Reply:
FWIW, I think the correct indent is halfway between the original and Jireh's suggestion, but I may be wrong (it's hard to get whitespace right in the github suggestion view, since everything shows in a non-monospaced font).